auth/src/main/java/org/aerogear/mobile/security/AbstractSecurityCheckExecutor.java

```diff
@@ -5,76 +5,82 @@
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
+import org.aerogear.mobile.core.executor.AppExecutors;
 import org.aerogear.mobile.core.metrics.MetricsService;
 
 import java.util.Collection;
 import java.util.HashSet;
 
+import static org.aerogear.mobile.core.utils.SanityCheck.nonNull;
+
 /**
  * Base class for security check executors.
  */
 abstract class AbstractSecurityCheckExecutor<T extends AbstractSecurityCheckExecutor> {
```

Comment:
Here you should document the @param T, ie:
`@param <T> the real executor type to be returned to chain the adding of checks to be executed`